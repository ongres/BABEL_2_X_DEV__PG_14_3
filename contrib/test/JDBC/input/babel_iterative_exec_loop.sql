-- simple loop
DECLARE @Counter INT
SET @Counter=1
WHILE ( @Counter <= 10)
BEGIN
	SELECT @Counter
	SET @Counter  = @Counter  + 1
END
GO

-- continue
DECLARE @Counter INT 
SET @Counter=1
WHILE ( @Counter <= 10)
BEGIN
	if @Counter / 2 = 1
	BEGIN
		SET @Counter  = @Counter  + 1
		CONTINUE
	END
	SELECT @Counter 
	SET @Counter  = @Counter  + 1
END
GO

-- break
DECLARE @Counter INT 
SET @Counter=1
WHILE ( @Counter <= 10)
BEGIN
	if @Counter = 7
		BREAK
	SELECT @Counter 
	SET @Counter  = @Counter  + 1
END
GO

-- nested loop
DECLARE @Counter1 INT 
DECLARE @Counter2 INT 
SET @Counter1 = 1
SET @Counter2 = 10
WHILE ( @Counter1 <= 5)
BEGIN
	SELECT @Counter1
	WHILE ( @Counter2 < 13)
	BEGIN
		SELECT @Counter2
		SET @Counter2  = @Counter2  + 1
	END
	SET @Counter1  = @Counter1  + 1
END
GO

-- test break within try-catch block
DECLARE @i INT
SET @i = 1
WHILE (@i < 3)
BEGIN
	BEGIN TRY
		BREAK
	END TRY
	BEGIN CATCH
	END CATCH
	SET @i = @i + 1
END 
SELECT @i
GO

DECLARE @i INT
SET @i = 1
WHILE (@i < 3)
BEGIN
	BEGIN TRY
		SELECT 1/0
	END TRY
	BEGIN CATCH
		BREAK
	END CATCH
	SET @i = @i + 1
END
SELECT @i
GO

-- test continue within try-catch block
DECLARE @i INT
SET @i = 1
WHILE (@i < 3)
BEGIN
	SELECT @i
	SET @i = @i + 1
	BEGIN TRY
		CONTINUE
	END TRY
	BEGIN CATCH
	END CATCH
	SET @i = @i + 2
END
GO

DECLARE @i INT
SET @i = 1
WHILE (@i < 3)
BEGIN
	SELECT @i
	SET @i = @i + 1
	BEGIN TRY
		SELECT 1/0
	END TRY
	BEGIN CATCH
		CONTINUE
	END CATCH
	SET @i = @i + 2
END
GO
