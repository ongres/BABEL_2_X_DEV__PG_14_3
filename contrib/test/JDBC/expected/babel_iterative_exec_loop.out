-- simple loop
DECLARE @Counter INT
SET @Counter=1
WHILE ( @Counter <= 10)
BEGIN
	SELECT @Counter
	SET @Counter  = @Counter  + 1
END
GO
~~START~~
int
1
~~END~~

~~START~~
int
2
~~END~~

~~START~~
int
3
~~END~~

~~START~~
int
4
~~END~~

~~START~~
int
5
~~END~~

~~START~~
int
6
~~END~~

~~START~~
int
7
~~END~~

~~START~~
int
8
~~END~~

~~START~~
int
9
~~END~~

~~START~~
int
10
~~END~~


-- continue
DECLARE @Counter INT 
SET @Counter=1
WHILE ( @Counter <= 10)
BEGIN
	if @Counter / 2 = 1
	BEGIN
		SET @Counter  = @Counter  + 1
		CONTINUE
	END
	SELECT @Counter 
	SET @Counter  = @Counter  + 1
END
GO
~~START~~
int
1
~~END~~

~~START~~
int
4
~~END~~

~~START~~
int
5
~~END~~

~~START~~
int
6
~~END~~

~~START~~
int
7
~~END~~

~~START~~
int
8
~~END~~

~~START~~
int
9
~~END~~

~~START~~
int
10
~~END~~


-- break
DECLARE @Counter INT 
SET @Counter=1
WHILE ( @Counter <= 10)
BEGIN
	if @Counter = 7
		BREAK
	SELECT @Counter 
	SET @Counter  = @Counter  + 1
END
GO
~~START~~
int
1
~~END~~

~~START~~
int
2
~~END~~

~~START~~
int
3
~~END~~

~~START~~
int
4
~~END~~

~~START~~
int
5
~~END~~

~~START~~
int
6
~~END~~


-- nested loop
DECLARE @Counter1 INT 
DECLARE @Counter2 INT 
SET @Counter1 = 1
SET @Counter2 = 10
WHILE ( @Counter1 <= 5)
BEGIN
	SELECT @Counter1
	WHILE ( @Counter2 < 13)
	BEGIN
		SELECT @Counter2
		SET @Counter2  = @Counter2  + 1
	END
	SET @Counter1  = @Counter1  + 1
END
GO
~~START~~
int
1
~~END~~

~~START~~
int
10
~~END~~

~~START~~
int
11
~~END~~

~~START~~
int
12
~~END~~

~~START~~
int
2
~~END~~

~~START~~
int
3
~~END~~

~~START~~
int
4
~~END~~

~~START~~
int
5
~~END~~


-- test break within try-catch block
DECLARE @i INT
SET @i = 1
WHILE (@i < 3)
BEGIN
	BEGIN TRY
		BREAK
	END TRY
	BEGIN CATCH
	END CATCH
	SET @i = @i + 1
END 
SELECT @i
GO
~~START~~
int
1
~~END~~


DECLARE @i INT
SET @i = 1
WHILE (@i < 3)
BEGIN
	BEGIN TRY
		SELECT 1/0
	END TRY
	BEGIN CATCH
		BREAK
	END CATCH
	SET @i = @i + 1
END
SELECT @i
GO
~~START~~
int
1
~~END~~


-- test continue within try-catch block
DECLARE @i INT
SET @i = 1
WHILE (@i < 3)
BEGIN
	SELECT @i
	SET @i = @i + 1
	BEGIN TRY
		CONTINUE
	END TRY
	BEGIN CATCH
	END CATCH
	SET @i = @i + 2
END
GO
~~START~~
int
1
~~END~~

~~START~~
int
2
~~END~~


DECLARE @i INT
SET @i = 1
WHILE (@i < 3)
BEGIN
	SELECT @i
	SET @i = @i + 1
	BEGIN TRY
		SELECT 1/0
	END TRY
	BEGIN CATCH
		CONTINUE
	END CATCH
	SET @i = @i + 2
END
GO
~~START~~
int
1
~~END~~

~~START~~
int
2
~~END~~

